code/game/objects/structures/watercloset.dm

```diff
@@ -471,3 +656,77 @@
 	icon_state = "puddle-splash"
 	..()
 	icon_state = "puddle"
+
+
+//////////////////////////////////
+//		Bathroom Fixture Items	//
+//////////////////////////////////
+
+
+/obj/item/mounted/shower
+	name = "shower fixture"
+	desc = "A self-adhering shower fixture. Simply stick to a wall, no plumber needed!"
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "shower"
+	item_state = "buildpipe"
+
+/obj/item/mounted/shower/try_build(turf/on_wall, mob/user, proximity_flag)
+	//overriding this because we don't care about other items on the wall, but still need to do adjacent checks
+	if(!on_wall || !user)
+		return
+	if(proximity_flag != 1) //if we aren't next to the wall
+		return
+	if(!(get_dir(on_wall,user) in cardinal))
+		to_chat(user, "<span class='rose'>You need to be standing next to a wall to place \the [src].</span>")
```

Comment:
That was copied right from mountable code. I didn't realize rose was a span either, but I'll change it for everything to maintain consistency.